pkg/build/builder/util_linux.go

```diff
@@ -5,15 +5,22 @@ import (
 	"os"
 
 	"github.com/opencontainers/runc/libcontainer/cgroups"
-
 	s2iapi "github.com/openshift/source-to-image/pkg/api"
 )
 
 // GetCGroupLimits returns a struct populated with cgroup limit values gathered
 // from the local /sys/fs/cgroup filesystem.  Overflow values are set to
 // math.MaxInt64.
 func GetCGroupLimits() (*s2iapi.CGroupLimits, error) {
-	byteLimit, err := readInt64("/sys/fs/cgroup/memory/memory.limit_in_bytes")
+	// see https://git.kernel.org/pub/scm/linux/kernel/git/tj/cgroup.git/tree/Documentation/admin-guide/cgroup-v2.rst
+	// for list of cgroupv2 files to try, but Nalin relayed that examination of the crun and runc code that 'memory.high'
+	// is not used.
+	file := "/sys/fs/cgroup/memory/memory.limit_in_bytes"
+	if cgroups.IsCgroup2UnifiedMode() {
+		file = "/sys/fs/cgroup/memory.max"
```

Comment:
Sorry I haven't raised this concern earlier, but here it is.

cgroup v2 does not have "memory.max" file in the top-level cgroup **on the host**, so this won't work **on the host** (unless you add a fallback to assume the limit is `max` or somesuch).

If this is done inside a container, this check will work, as the container has its own cgroup mounted to /sys/fs/cgroup, not the top one.

Hope that helps.

Reply:
thanks @kolyshkin - that clarification, and the additional host level verification details you added in the associated bugzilla, should help with the confusion @wewang58 was having in trying to verify a proper set up to verify.

As it turns out, this logic is in fact running in a container running in the build pod.

@adambkaplan I think we are ready for the "official" lgtm so we can get this merged, unless you had any other questions.

Reply:
It's all good then.